### PR TITLE
Mve2 pong api gameover

### PR DESCRIPTION
**Pull request by romann-broque**

### CODE REVIEW

File: back/transcendence_django/back_game/monitor/monitor.py [hunk 1]
```diff
@@ -47,40 +41,61 @@ def get_channel_from_username(self, username):
...
+        for arena in arenas.values():
+            if arena.status == STARTED and arena.is_empty():
+                arena.end_of_game()
+            if arena.status == OVER:
+                logger.info(f"Game over in arena {arena.id}")
+                await self.gameOver(arenas, arena)
+                break
+
+    async def run_game_loop(self, arenas):
+        while len(arenas) > 0:
```
Comment (romann-broque) [inline back/transcendence_django/back_game/monitor/monitor.py:82]: replace by 'arenas'

File: front/src/app/pages/home-page/home-page.component.ts [hunk 2]
```diff
@@ -20,9 +22,12 @@ export class HomePageComponent implements OnInit {
...
-  ngOnInit(): void {
-    this.userService.getUsername().subscribe(data => {
-      this.welcome = `Welcome, ${data.username}!`;
-    });
+  async ngOnInit(): Promise<void> {
+    await this.userService.whenUserDataLoaded();
+    this.welcome = `Welcome, ${this.userService.getUsername()}`;
+ }
+
+ public getLocalGameUrl(): string {
```
Comment (romann-broque) [inline front/src/app/pages/home-page/home-page.component.ts:30]: remove this method

File: front/src/app/services/web-socket/web-socket.service.ts [hunk 3]
```diff
@@ -61,13 +88,39 @@ export class WebSocketService {
...
+          subject.next(data.arena);
+        }
+      });
     } else {
       console.log('WebSocket is not open when trying to join arena');
     }
+    return subject.asObservable();
   }
 
   public leave(): void {
```
Comment (romann-broque) [inline front/src/app/services/web-socket/web-socket.service.ts:126]: remove the useless leave method

**Review by romann-broque — COMMENTED** [review]
Noice !

**Review by opaleboreale — APPROVED** [review]

**Pull request merged**